State.cpp

```diff
@@ -0,0 +1,248 @@
+#include "BreathData.h"
+#include "State.h"
+#include "Constants.h"
+#include "Display.h"
+#include "Flow.h"
+#include "Pressure.h"
+#include "Valve.h"
+#include "ProportionalValve.h"
+#include "AlarmManager.h"
+#include "State.h"
+
+// Initialize singleton instances
+OffState* OffState::instance = 0;
+InsStateVC* InsStateVC::instance = 0;
+InsHoldStateVC* InsHoldStateVC::instance = 0;
+ExpStateVC* ExpStateVC::instance = 0;
+PeepStateVC* PeepStateVC::instance = 0;
+ExpHoldStateVC* ExpHoldStateVC::instance = 0;
```

Comment:
All of these should go away, as per my comment on `OffState::begin`

Reply:
They're still here.  Singletons don't need global pointers; they need global objects:
```
ExpStateVC ExpStateVC::instance;
```
Or better yet, make it a singleton static member function:
```
ExpStateVC& ExpStateVC::singleton()  // Static
{
    ExpStateVC theSingleton;
    return theSingleton;
}
```